### ISSUE TITLE
support `filter` argument in cache decorator

### ISSUE THREAD

**Issue by mattshin** [long2ice/fastapi-cache#15]
It would be nice to have an optional parameter such as `filter` in the `@cache` decorator that took as input a python callable w/ the wrapped function's `*args` and `**kwargs` and returned a boolean as to whether or not to cache.

The default could just be always true e.g. `lambda x: True` but users may want more nuanced control, based on the parameters to the function itself. In my particular case, I can't quite use `cache-control` header because i want to determine whether or not to cache on the _server side_  based on query params.

If I get some time, I can jam on this and submit a PR, if you think it could be worthwhile?

**Comment by long2ice** [comment]
Does `key_builder` param satisfy you?

**Comment by mattshin** [comment]
not quite, although I _technically_ could use it to hack a solution by returning a key I know has not been cached. Ideally, the filter callable is called before a key is even built, and if the filter returns false, then the entire caching process (both get and set) is not attempted.

**Comment by mjpieters** [comment]
I was already thinking the key builder should be able to flag a request as not cacheable, either by returning `None` or raising an exception here.
